core/templates/dev/head/collection_editor/CollectionEditorStateService.js

```diff
@@ -0,0 +1,185 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to maintain the state of a single collection shared
+ * throughout the collection editor. This service provides functionality for
+ * retrieving the collection, saving it, and listening for changes. The service
+ * also maintains a list of all skills stored within the collection.
+ */
+
+oppia.constant(
+  'COLLECTION_EDITOR_INITIALIZED_COLLECTION',
+  'collectionEditorInitializedCollection');
+oppia.constant(
+  'COLLECTION_EDITOR_UPDATED_COLLECTION', 'collectionEditorUpdatedCollection');
+
+oppia.factory('CollectionEditorStateService', [
+  '$rootScope', 'alertsService', 'CollectionObjectFactory',
+  'SkillListObjectFactory', 'UndoRedoService',
+  'WritableCollectionBackendApiService',
+  'COLLECTION_EDITOR_INITIALIZED_COLLECTION',
```

Comment:
Think it's OK to shorten to EVENT_COLLECTION_INITIALIZED, EVENT_COLLECTION_UPDATED? We're always in the collection editor anyway...